jormungandr/src/network/p2p/topology.rs

```diff
@@ -2,171 +2,207 @@
 //!
 
 use crate::{
-    network::p2p::{layers::PreferredListLayer, Address, Gossips, Policy, PolicyConfig},
+    network::p2p::{Address, Gossip, Gossips, Quarantine},
     settings::start::network::Configuration,
 };
-use poldercast::{
-    custom_layers,
-    poldercast::{Cyclon, Rings, Vicinity},
-    NodeProfile, PolicyReport, StrikeReason, Topology,
-};
-use rand_chacha::ChaChaRng;
+use jormungandr_lib::time::SystemTime;
+use poldercast::{Profile, Topology};
+use serde::Serializer;
 use tokio::sync::RwLock;
-use tracing::{span, Level, Span};
+use tracing::instrument;
 
-pub struct View {
-    pub self_node: NodeProfile,
-    pub peers: Vec<Address>,
-}
+use super::{topic, NodeId};
 
-/// object holding the P2pTopology of the Node
-pub struct P2pTopology {
-    lock: RwLock<Topology>,
-}
+use std::hash::{Hash, Hasher};
+use std::sync::Arc;
 
-/// Builder object used to initialize the `P2pTopology`
-struct Builder {
-    topology: Topology,
-    span: Span,
+fn serialize_display<T: std::fmt::Display, S>(item: &T, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    serializer.serialize_str(&item.to_string())
 }
 
-impl Builder {
-    /// Create a new topology for the given node profile
-    fn new(node: poldercast::NodeProfile, span: Span) -> Self {
-        Builder {
-            topology: Topology::new(node),
-            span,
-        }
-    }
+#[derive(Eq, Clone, Serialize, Debug)]
+pub struct ProfileInfo {
+    #[serde(serialize_with = "serialize_display")]
```

Comment:
Protip: You can use [`serde_with`](https://crates.io/crates/serde_with) to avoid reinventing some common helpers.